Multiplication on closed intervals of rational numbers (#1351)
UniMath/agda-unimath: src/elementary-number-theory/minimum-rational-numbers.lagda.md

# Minimum on the rational numbers

```agda
module elementary-number-theory.minimum-rational-numbers where
```

<details><summary>Imports</summary>

```agda
open import elementary-number-theory.decidable-total-order-rational-numbers
open import elementary-number-theory.inequality-rational-numbers
open import elementary-number-theory.rational-numbers
open import elementary-number-theory.strict-inequality-rational-numbers

open import foundation.coproduct-types
open import foundation.identity-types
open import foundation.transport-along-identifications

open import order-theory.decidable-total-orders
```

</details>

## Idea

> We define the operation of minimum

Reviewer: ```suggestion
We define the operation of {{#concept "minimum" Disambiguation="of pairs of rational numbers" Agda=min-ℚ}}
```

([greatest lower bound](order-theory.greatest-lower-bounds-posets.md)) for the
[rational numbers](elementary-number-theory.rational-numbers.md).

## Definition

```agda
min-ℚ : ℚ → ℚ → ℚ
min-ℚ = min-Decidable-Total-Order ℚ-Decidable-Total-Order
```

## Properties

### Associativity of `min-ℚ`

```agda
associative-min-ℚ : (x y z : ℚ) → min-ℚ (min-ℚ x y) z ＝ min-ℚ x (min-ℚ y z)
associative-min-ℚ =
  associative-min-Decidable-Total-Order ℚ-Decidable-Total-Order
```

> ### Commutativity of `min-ℚ`

Reviewer: ```suggestion
### Commutativity of the minimum operation
```

```agda
commutative-min-ℚ : (x y : ℚ) → min-ℚ x y ＝ min-ℚ y x
commutative-min-ℚ =
  commutative-min-Decidable-Total-Order ℚ-Decidable-Total-Order
```

> ### `min-ℚ` is idempotent

Reviewer: ```suggestion
### The minimum operation is idempotent
```

```agda
idempotent-min-ℚ : (x : ℚ) → min-ℚ x x ＝ x
idempotent-min-ℚ =
  idempotent-min-Decidable-Total-Order ℚ-Decidable-Total-Order
```

### The minimum is a lower bound

```agda
leq-left-min-ℚ : (x y : ℚ) → min-ℚ x y ≤-ℚ x
leq-left-min-ℚ = leq-left-min-Decidable-Total-Order ℚ-Decidable-Total-Order

leq-right-min-ℚ : (x y : ℚ) → min-ℚ x y ≤-ℚ y
leq-right-min-ℚ = leq-right-min-Decidable-Total-Order ℚ-Decidable-Total-Order
```

> ### If `a ≤ b`, `min a b = a`

Reviewer: ```suggestion
### If `a` is less than or equal to `b`, then the minimum of `a` and `b` is `a`
```

# Minimum on the rational numbers

```agda
module elementary-number-theory.minimum-rational-numbers where
```

<details><summary>Imports</summary>

```agda
open import elementary-number-theory.decidable-total-order-rational-numbers
open import elementary-number-theory.inequality-rational-numbers
open import elementary-number-theory.rational-numbers
open import elementary-number-theory.strict-inequality-rational-numbers

open import foundation.coproduct-types
open import foundation.identity-types
open import foundation.transport-along-identifications

open import order-theory.decidable-total-orders
```

</details>

## Idea

> We define the operation of
> {{#concept "minimum" Disambiguation="of pairs of rational numbers" Agda=min-ℚ}}
> ([greatest lower bound](order-theory.greatest-lower-bounds-posets.md)) for the
> [rational numbers](elementary-number-theory.rational-numbers.md).

Reviewer: ```suggestion
The {{#concept "minimum" Disambiguation="of pairs of rational numbers" Agda=min-ℚ}} of two [rational numbers](elementary-number-theory.rational-numbers.md) is the [smallest](elementary-number-theory.inequality-rational-numbers.md) rational number of the two. This is the [binary greatest lower bound](order-theory.greatest-lower-bounds-posets.md) in the [total order of rational numbers](elementary-number-theory.decidable-total-order-rational-numbers.md).
```
